python/sdist/amici/de_model.py

```diff
@@ -713,3 +714,21 @@ def __eq__(self, other):
         return self.get_val() == other.get_val() and (
             self.get_initial_value() == other.get_initial_value()
         )
+
+    def triggers_at_fixed_timepoint(self) -> bool:
+        """Check whether the event triggers at a (single) fixed time-point."""
+        t_root = sp.solve(self.get_val(), amici_time_symbol)
```

Comment:
Does it make sense to cache the value of t_root at initialization? `get_trigger_time` alone computes this twice.  Would also reduce some redundancy.

Reply:
Can be done. I don't think computing it twice matters performance-wise, but the redundancy is not so nice, agreed.